### PR TITLE
BUAN framework documentation

### PR DESCRIPTION
**Pull request by BramshQamar**
Added BUAN_flow.rst file for the documentation of Bundle Analytics pipeline CLIs.

### CODE REVIEW

**Comment by BramshQamar** [comment]
Hi @skoudoro and @Garyfallidis,

This is the new PR. I have closed the previous PR.

**Comment by skoudoro** [comment]
Hi @BramshQamar,

is this PR ready for a review?

**Comment by BramshQamar** [comment]
> Hi @BramshQamar,
> 
> is this PR ready for a review?

Hi @skoudoro, 

Just waiting for NIH figshare approval then will add the link of test data. Other than that it's ready.

**Comment by BramshQamar** [comment]
> > Hi @BramshQamar,
> > is this PR ready for a review?
> 
> Hi @skoudoro,
> 
> Just waiting for NIH figshare approval then will add the link of test data. Other than that it's ready.

okay now ready with all links @skoudoro

File: doc/interfaces/BUAN_flow.rst [hunk 1]
```diff
@@ -0,0 +1,85 @@
+.. _BUAN_flow:
+
+=====================================================================================
+   Tutorial for BUndle ANalytics (BUAN) framework using DIPY Workflow command lines
```
Comment (skoudoro) [inline doc/interfaces/BUAN_flow.rst:4]: No spaces before to Tutorial. Need to be removed

File: doc/interfaces/BUAN_flow.rst [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+=====================================================================================
```
Comment (skoudoro) [inline doc/interfaces/BUAN_flow.rst:5]: should be the same size as the title.  same for the above

File: doc/interfaces/BUAN_flow.rst [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+For generating bundle profile data (saved as .h5 files):
+You must have downloaded bundles folder of 30 atlas bundles and subjects folder
+with PPMI data derivatives.
+
+NOTE: Make sure all output folders are empty and do not get overridden.
+Following workflows require specific input directory structure.
+
+Create an out_dir folder (eg: bundle_profiles)::
+
+    mkdir bundle_profiles
```
Comment (skoudoro) [inline doc/interfaces/BUAN_flow.rst:41]: Can you add `$` in front. e.g `$ mkdir bundle_profiles`. Same comment for all command line below

File: doc/interfaces/BUAN_flow.rst [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+Run thee following workflow::
+
+    dipy_buan_profiles bundles/ subjects_small/ --out_dir "bundle_profiles"
```
Comment (skoudoro) [inline doc/interfaces/BUAN_flow.rst:45]: Can you add `$` in front.

File: doc/interfaces/BUAN_flow.rst [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+
+For running Linear Mixed Models (LMM) on generated .h5 files from the previous
+step:
+
+Create an out_dir folder (eg: lmm_plots)::
```
Comment (skoudoro) [inline doc/interfaces/BUAN_flow.rst:51]: ` ``out_dir`` `

File: doc/interfaces/BUAN_flow.rst [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+    mkdir lmm_plots
```
Comment (skoudoro) [inline doc/interfaces/BUAN_flow.rst:53]: Can you add `$` in front.

File: doc/interfaces/BUAN_flow.rst [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+Progression Markers Initiative (PPMI) Data Derivatives. It can be downloaded
+from the link below::
+
+     https://doi.org/10.35092/yhjc.12098397
+
+There are two parts of Bundle Analytics group comparison framework,
+bundle profile analysis and bundle shape similarity.
+
+-------------------------------------------------------------------
+      We first start with group comparison of bundle profiles
```
Comment (skoudoro) [inline doc/interfaces/BUAN_flow.rst:29]: I need to check if these spaces do not generate an error. But you can do it too! 
```
$ cd doc
$ pip install -U -r doc-requirements.txt
$ make -C . clean && make -C . html-after-examples
$ open build/html/index.html
```

File: doc/interfaces/BUAN_flow.rst [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+For running Linear Mixed Models (LMM) on generated .h5 files from the previous
+step:
+
+Create an out_dir folder (eg: lmm_plots)::
+
+    mkdir lmm_plots
+
+and run the following workflow::
+
+    dipy_buan_lmm "bundle_profiles/*" --out_dir "lmm_plots"
```
Comment (skoudoro) [inline doc/interfaces/BUAN_flow.rst:57]: same comment as above

File: doc/interfaces/BUAN_flow.rst [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+-------------------------------------------------------------------------------
+  Calculating shape similarity of same type of bundles across the population
```
Comment (skoudoro) [inline doc/interfaces/BUAN_flow.rst:60]: same comment as above

File: doc/interfaces/BUAN_flow.rst [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+-------------------------------------------------------------------------------
+
+Create an out_dir folder (eg: sm_plots)::
```
Comment (skoudoro) [inline doc/interfaces/BUAN_flow.rst:63]: same comment as above for out_dir

File: doc/interfaces/BUAN_flow.rst [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+    mkdir sm_plots
```
Comment (skoudoro) [inline doc/interfaces/BUAN_flow.rst:65]: same comment as above

File: doc/interfaces/BUAN_flow.rst [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+
+run following workflow::
+
+    dipy_buan_shapes subjects/ --out_dir "sm_plots"
```
Comment (skoudoro) [inline doc/interfaces/BUAN_flow.rst:69]: same comment as above

File: doc/interfaces/BUAN_flow.rst (hunk 3 above)
Comment (BramshQamar) [reply doc/interfaces/BUAN_flow.rst]: with $ it generates something like this:

:math:` mkdir lmm_plots

Comment (BramshQamar) [reply doc/interfaces/BUAN_flow.rst]: with $
` `:math:`mkdir bundle_profiles `

without $
`mkdir bundle_profiles`

File: doc/interfaces/BUAN_flow.rst [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+Create an ``out_dir`` folder (eg: sm_plots)::
+
+    mkdir sm_plots
+
+run following workflow::
+
+    dipy_buan_shapes subjects/ --out_dir "sm_plots"
+
+
+For more information about each command line, you can got to :ref:`workflows_reference`.
```
Comment (Garyfallidis) [inline doc/interfaces/BUAN_flow.rst:72]: go to?

File: doc/interfaces/BUAN_flow.rst (hunk 3 above)
Comment (BramshQamar) [reply doc/interfaces/BUAN_flow.rst:41]: Screen Shot 2020-04-15 at 5 42 13 PM

Comment (skoudoro) [reply doc/interfaces/BUAN_flow.rst:41]: ok, I see, thank you for this info

File: doc/interfaces/BUAN_flow.rst [hunk 14]
```diff
@@ -68,18 +72,49 @@ run following workflow::
...
     dipy_buan_shapes subjects/ --out_dir "sm_plots"
 
+--------------------------------------
+Reproducing results on larger dataset:
+--------------------------------------
+
+Complete dataset DIPY Processed Parkinson's Progression Markers Initiative (PPMI)
+Data Derivatives. It can be downloaded from the link below::
+
+     insert link when gets publishes
```
Comment (skoudoro) [inline doc/interfaces/BUAN_flow.rst:82]: Does it mean we should wait for the publication before merging this PR?

Comment (BramshQamar) [reply doc/interfaces/BUAN_flow.rst:82]: This is for large data set.
I have published it from my side. waiting for NIH to publish.

Comment (skoudoro) [reply doc/interfaces/BUAN_flow.rst:87]: ok

File: doc/interfaces/BUAN_flow.rst [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+
+For this tutorial we will be using a test sample of DIPY Processed Parkinson's
+Progression Markers Initiative (PPMI) Data Derivatives. It can be downloaded
+from the link below
+
+     `<https://doi.org/10.35092/yhjc.12098397>`_
+
+.. note::
+
+    If you prefer to run experiments on complete dataset please see the
```
Comment (Garyfallidis) [inline doc/interfaces/BUAN_flow.rst:28]: on the complete dataset to reproduce the paper

File: doc/interfaces/BUAN_flow.rst [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+    "Reproducing results on larger dataset" section at the end of the page for more
```
Comment (Garyfallidis) [inline doc/interfaces/BUAN_flow.rst:29]: end of page.

File: doc/interfaces/BUAN_flow.rst [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+    information.
+
+There are two parts of Bundle Analytics group comparison framework,
+bundle profile analysis and bundle shape similarity.
```
Comment (Garyfallidis) [inline doc/interfaces/BUAN_flow.rst:33]: similarity analysis

File: doc/interfaces/BUAN_flow.rst [hunk 18]
```diff
@@ -0,0 +1,153 @@
...
+
+.. figure:: https://github.com/dipy/dipy_data/blob/master/AF_L_fa.png?raw=true
+    :width: 70 %
+    :alt: alternate text
+    :align: center
+
+    Result plot for left arcuate fasciculus (AF_L) on FA measure
+
+--------------------------------------------------------------------------
+Calculating shape similarity of same type of bundles across the population
```
Comment (Garyfallidis) [inline doc/interfaces/BUAN_flow.rst:83]: Shape similarity of specific bundles across the populations

**Pull request merged**